### PR TITLE
Added Security Center, Unittest, builds and refactored

### PR DESCRIPTION
**Pull request by UncleBats**
- Added security center so it will work in the same way for the new continuous export to the event hub.
- Refactored into more flow orientated structure, compared to the previous technical structure.
- Replaced the dependent HOME environmental variable with the FunctionAppDirectory from the function context. (HOME environmental existed on the hosted build server so tests would fail otherwise)
- Added some simple translation unit-test to be sure I didn't knock anything over
- Added YAML build pipeline
- Fixed some bugs and unused code

### CODE REVIEW

**Comment by sebastus** [comment]
Thanks very much for the contribution, @ErickSegaar. As it's changing/adding quite a number of files, it'll take a bit to review it all. Any recommendations on the best way to proceed through it all?

**Comment by ErickSegaar** [comment]
@sebastus My thoughts on the matter are the following:

- Refactored into event hub processing eg (activity, diagnostic, metrics). Shouldn't be changed much, naming is consistent with original except added the type
- Changed the way the categories.json worked. There was code depending on a HOME variable, that gave some trouble with debugging and testing local and using the build server. Didn't like having specific code just for testing in the solution so changed it in SplunkEventMessages Ctor
- Added SecurityCenterLogs based upon the Diagnostic logs (didn't change your deployment template (using a terraform option instead of the azure template, should add additional security hub in the infra)
- Fixed the Metrics to use the metrics classes instead of the Diagnostic classes (   var azMonMsg = new AzMonDiagnosticLog(expandoRecord); ==>              var azMonMsg = new Metric(expandoRecord);)
- Added the Yaml build 
- I am running the total solution now with 2.7 mil events a minute works like a charm, haven't done anything with the Windows and Linux VM so didn't test that.
- Sorry for the big dump, had a lot of trouble with the HOME variable in the end and basically squashed my commits, could have better only squashed the changes for the variable. Wrote a little Down the rabbit hole blog on it to :).

**Comment by sebastus** [comment]
Erick - would you contact me offline? golive@microsoft.com.

**Review by sebastus — COMMENTED** [review]
General comment - all source files require inclusion of the copyright/license statement that you can find in (for example) the unmodified runner.cs.

File: AzureFunctionForSplunk/.gitignore [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ## files generated by popular Visual Studio add-ons.
 
 # Azure Functions localsettings file
-local.settings.json
+*local.settings.json
```
Comment (sebastus) [inline AzureFunctionForSplunk/.gitignore]: My local settings has secrets in it - a security risk. Is it necessary to check it in?

Comment (ErickSegaar) [reply AzureFunctionForSplunk/.gitignore]: The * doesn't comment it out so it is still in effect, but I don't see any reason to write it that way nor do I remember why I did that. Let's change it back

File: AzureFunctionForSplunk/ActivityLogs/AzMonActivityLog.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
+﻿//
```
Comment (sebastus) [inline AzureFunctionForSplunk/ActivityLogs/AzMonActivityLog.cs]: This file seems redundant with the addition of ActivityLogs/ActivityLog.cs.

Comment (ErickSegaar) [reply AzureFunctionForSplunk/ActivityLogs/AzMonActivityLog.cs]: Your right, forgotten to take them out after changes I think. I made the same mistake for metrics and diagnostics.

Comment (sebastus) [reply AzureFunctionForSplunk/ActivityLogs/AzMonActivityLog.cs]: Thanks.

File: AzureFunctionForSplunk/SecurityCenterLogs/SecurityLog.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            Message = message;
+
+            if (((IDictionary<string, object>)message).ContainsKey("AzureResourceId"))
+            {
+                ResourceId = message.AzureResourceId;
+            }
+            else
+            {
+                ResourceId = message.properties.resourceDetails.id;
```
Comment (sebastus) [inline AzureFunctionForSplunk/SecurityCenterLogs/SecurityLog.cs]: Can you post a raw example of a security center log record? The reason that a couple of the other log types look for both resourceId and resourceid is that over time the product team has published logs with slightly different spellings (case) of the thing. But it's not easy to find that sort of detail in any docs.

**Pull request closed without merging**